canonical-web-and-design/maas-ui
# feat(ui): amend KVMComposeForm to be compatible with RSD VM hosts

## Done

- Amended KVM's ComposeForm component to work with RSDs.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that clicking on `RSD` in the main navigation takes you to the React RSD list. 
- Go to an RSD details page
- Note that `make sampledata`d RSDs will have a bunch of KVM-like storage pools. This is incorrect - there should only be "Local" or "iSCSI" storage.
- Choose Compose from the action menu
- Check that instead of a rich pool select in the Storage section, it's just a normal select. It should always have the option "Local".
- Check that if the pod has iSCSI storage (`pod.capabilities.includes("iscsi_storage")`) there is also the option for "iSCSI".

## Fixes

Fixes #1641
